analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/BranchingProcessor.cs

```diff
@@ -30,24 +30,24 @@ namespace SonarAnalyzer.SymbolicExecution.Roslyn.OperationProcessors;
 internal abstract class BranchingProcessor<T> : MultiProcessor<T>
     where T : IOperationWrapper
 {
-    protected abstract SymbolicConstraint BoolConstraintFromOperation(ProgramState state, T operation, int visitCount);
-    protected abstract ProgramState LearnBranchingConstraint(ProgramState state, T operation, int visitCount, bool falseBranch);
+    protected abstract SymbolicConstraint BoolConstraintFromOperation(ProgramState state, T operation, bool isLoopCondition, int visitCount);
+    protected abstract ProgramState LearnBranchingConstraint(ProgramState state, T operation, bool isLoopCondition, int visitCount, bool falseBranch);
 
     protected virtual ProgramState PreProcess(ProgramState state, T operation) =>
         state;
 
     protected override ProgramState[] Process(SymbolicContext context, T operation)
     {
         var state = PreProcess(context.State, operation);
-        if (BoolConstraintFromOperation(state, operation, context.VisitCount) is { } constraint)
+        if (BoolConstraintFromOperation(state, operation, context.IsLoopCondition, context.VisitCount) is { } constraint)
```

Comment:
We can't because `context.State` and `state` are two different things and it would be confusing to send both, unfortunately.

If there will be a need for more stuff, I'll make some `record struct` with the arguments inside the `BranchingProcessor` class